packages/models/src/lib/category-config.unit.test.ts

```diff
@@ -1,29 +1,252 @@
+      categoryConfigSchema.parse({
+        slug: 'bug-prevention',
+        title: 'Bug prevention',
+        refs: [
+          {
+            plugin: 'eslint',
+            slug: 'no-magic-numbers',
+            type: 'audit',
+            weight: 1,
+          },
+        ],
+      } satisfies CategoryConfig),
+    ).not.toThrow();
+  });
+
+  it('should throw for an empty category', () => {
+    expect(() =>
+      categoryConfigSchema.parse({
+        slug: 'in-progress',
+        title: 'This category is empty for now',
+        refs: [],
+      } satisfies CategoryConfig),
+    ).toThrow('In a category there has to be at least one ref');
   });
 
-  it('should throw if only refs with weight 0 are included', () => {
-    const categoryCfg = categoryConfig();
-    const ref = { ...categoryCfg.refs[0], weight: 0 };
-    categoryCfg.refs = [ref];
+  it('should throw for duplicate category references', () => {
+    expect(() =>
+      categoryConfigSchema.parse({
+        slug: 'jest',
+        title: 'Jest results',
+        refs: [
+          {
+            plugin: 'jest',
+            slug: 'jest-unit-tests',
+            type: 'audit',
+            weight: 1,
+          },
+          {
+            plugin: 'jest',
+            slug: 'jest-unit-tests',
+            type: 'audit',
+            weight: 2,
+          },
+        ],
+      } satisfies CategoryConfig),
+    ).toThrow('audit or group refs are duplicates');
+  });
+
+  it('should throw for a category with only zero-weight references', () => {
+    expect(() =>
+      categoryConfigSchema.parse({
+        slug: 'informational',
+        title: 'This category is informational',
+        refs: [
+          {
+            plugin: 'eslint',
+            slug: 'functional/immutable-data',
+            type: 'audit',
+            weight: 0,
+          },
+          {
+            plugin: 'lighthouse',
+            slug: 'lighthouse-experimental',
+            type: 'group',
+            weight: 0,
+          },
+        ],
+      } satisfies CategoryConfig),
+    ).toThrow('In a category there has to be at least one ref with weight > 0');
+  });
+});
+
+describe('categoriesSchema', () => {
+  it('should accept a valid category array', () => {
+    expect(() =>
+      categoriesSchema.parse([
+        {
+          slug: 'bug-prevention',
+          title: 'Bug prevention',
+          refs: [
+            {
+              plugin: 'eslint',
+              slug: 'no-magic-numbers',
+              type: 'audit',
+              weight: 1,
+            },
+          ],
+        },
+        {
+          slug: 'code-style',
+          title: 'Code style',
+          refs: [
+            {
+              plugin: 'eslint',
+              slug: 'consistent-test-it',
+              type: 'audit',
+              weight: 1,
+            },
+          ],
+        },
+      ] satisfies CategoryConfig[]),
+    ).not.toThrow();
+  });
+
+  it('should throw for an empty category array', () => {
```

Comment:
Perhaps categories should be optional, i.e. could be missing or empty array? :thinking: Motivation would be to simplify onboarding.

Out of scope of this PR.